armi/reactor/reactors.py

```diff
@@ -2262,42 +2251,54 @@ def processLoading(self, cs):
                 "Please make sure that this is intended and not a input error."
             )
 
-        # if we have these params loaded from the database do not overwrite,
-        # otherwise initialize them
-        if not self.p.referenceBlockAxialMesh:
+        # TODO: There is an inconsistency with makeAxialSnapList when detailedAxialExpansion: False
+        # In a standard run from the blueprints:
+        #     - All assemblies (blueprints included) have their b.p.topIndex
+        #       set in reference to self.refAssem.getAxialMesh().
+        # In a database load run:
+        #     - All assemblies in the core have their b.p.topIndex as set in the standard run.
+        #     - The blueprints assemblies have the b.p.topIndex set in reference to the assembly with the finest mesh.
+        #       This guarantees functionality of makeAxialSnapList, but, if the assembly with the finest mesh is
+        #       not a fuel assembly, assembly.setBlockMesh may produce unexpected results.
```

Comment:
but if finest mesh is not fuel, detailedAxialExpansion wont work anyway since all assemblies must have submesh of ref assembly. I think this inconsistency is outside the scope of this PR and more to do with determining the refAssembly by the centermost assembly. Lets make an issue describing that we need a better way to determine ref assembly and delete this todo, or at least reduce it.

Reply:
Removed TODO here https://github.com/terrapower/armi/pull/797/commits/7df5b13b56fe5ec2afb5ffa07487001a816833d2

Will create issue to document the inconsistency and if/how we want to go about resolving it.